### ISSUE TITLE
Cannot get Absolute Series Scanner to show up in the settings

### ISSUE THREAD

**Issue by Hezkezl** [ZeroQI/Absolute-Series-Scanner#351]
## Platform

Operating system and version: Windows 10, Version: 10.0.18363 Build 18363
Plex version: 1.22.3.4392

## Expected Behavior
What I'm expecting to happen, is to have ASS show up as an option in the Settings -> Server -> Agents -> Shows list, but it does not appear there.

## Current Behavior
I only see "Personal Media Shows", "TheTVDB", and "The Movie Database" as options for the "Shows" listing. 
No log file is generated at `c:\users\NAME\AppData\Local\Plex Media Server\Logs`... no log files at all, let alone one specific to ASS.

## Steps to Reproduce

1. Created a /Scanners/ folder, and then inside that I created a /Series/ folder inside the appropriate location
2. Placed "Absolute Series Scanner.py" in my `C:\Users\NAME\AppData\Local\Plex Media Server\Scanners\Series` folder. https://cdn.discordapp.com/attachments/151525074644631553/835914585457557514/unknown.png
3. Restarted Plex Media Server
4. Open Plex, go to settings, cannot see Scanner listed in the options anywhere. https://cdn.discordapp.com/attachments/151525074644631553/835915704196268052/unknown.png 
5. Go to try to add a new library, but ASS doesn't appear as a possibility to use for the scanner there either. Only "Plex TV Series" and "Plex Series Scanner" shows up. https://cdn.discordapp.com/attachments/468978648460820490/835903133443948563/unknown.png

## Additional information

Restarted Plex multiple times, tried right-clicking Plex Media Server in my system tray, going to 'Open Plex...' and browsing to the settings on that, no dice. Tried the Plex App settings, no dice... This should be as simple as putting that .py file in the right folder and restarting Plex, but that doesn't work for me for some reason :(

**Comment by DJScias** [comment]
Hello there,

Considering Absolute Series Scanner (ASS) is not an agent (the agent would be [HAMA](https://github.com/ZeroQI/Hama.bundle), shown as HamaTV under Shows Agents) it will never show under Agents. That isn't a bug from what I can tell, since no scanner shows up there. It's never showed it for me here either, and I've been using this for years now.

Aside from the above. I unfortunately have no clue why ASS is not showing up as a scanner option since that -should- work (like in the img on 4.). So that's something someone else will probably have to answer/figure out.

However I would highly recommend you to get the Agent part of the duo (HAMA linked earlier) and then do a full restart of Plex and see if the two pieces make it all work? I don't see any use-case of using ASS and HAMA by themselves as they complement one another. Using them apart is like having a train with just wagons and without a locomotive.

**Comment by ZeroQI** [comment]
"Plex Series scanner" own scanner isn't showing in the agents section, why would "Absolute Series Scanner" ?
It does need to show as a scanner option in library option (point 5), and if it doesn't, location (point 4 looks good though) or rights are wrong

Check "Plex Series Scanner.log" or attach it, but i am pretty sure the plex service/user doesn't have the rights to that file...

**Comment by Hezkezl** [comment]
> 
> 
> Hello there,
> 
> Considering Absolute Series Scanner (ASS) is not an agent (the agent would be [HAMA](https://github.com/ZeroQI/Hama.bundle), shown as HamaTV under Shows Agents) it will never show under Agents. That isn't a bug from what I can tell, since no scanner shows up there. It's never showed it for me here either, and I've been using this for years now.
> 
> Aside from the above. I unfortunately have no clue why ASS is not showing up as a scanner option since that -should- work (like in the img on 4.). So that's something someone else will probably have to answer/figure out.
> 
> However I would highly recommend you to get the Agent part of the duo (HAMA linked earlier) and then do a full restart of Plex and see if the two pieces make it all work? I don't see any use-case of using ASS and HAMA by themselves as they complement one another. Using them apart is like having a train with just wagons and without a locomotive.

I actually am attempting to use both HAMA and ASS for this. I do have `AppData/Local/Plex Media Server/Plug-ins/Hama.bundle` installed as well, but I thought that since step 1 of HAMA's install guide says to use ASS, I thought I'd get that working first before trying to use the other one.

I very likely was getting my terminology backwards, with scanner/agent. But the main point being that when I go to add a new library of just Anime, and I set it to the 'TV Shows' type, then add folders, then the next tab is supposed to be where you select the scanner but it never showed ASS as an option. (HAMA doesn't show up as an agent either, with it being in the Plug-ins folder, but that's an issue for another time once I'm hopefully able to get this working)

I'm tempted to just completely uninstall Plex and do a brand new install, since that's usually how my luck goes with these problems :(

**Comment by ZeroQI** [comment]
in your library, you create a Series or movies library, add folder(s), select the scanner and (metadata) agent.
If ASS scanner is not showing, the content of the file is wrong, or file at wrong location (sometimes 2 locations for plex on windows)  or a right issue

Hama location should be: Plex Media Server/Plug-ins/Hama.bundle/Contents/Code/__init__.py

**Comment by Hezkezl** [comment]
I do have HAMA at `C:\Users\Jordan\AppData\Local\Plex Media Server\Plug-ins\Hama.bundle\Contents\Code`
Here's a screenshot showing what's in that location: ![1](https://user-images.githubusercontent.com/15794090/116121260-aee07300-a67d-11eb-9008-7b48e569a060.jpg)

I recorded my screen and showed both the HAMA folders and the ASS folders, then showing that Plex itself doesn't show them when I go to create a new series... I have zero idea what else to do to try to fix this :\ everything is in the place it's supposed to be, but Plex isn't recognizing anything. Here's the video: https://www.youtube.com/watch?v=QFmNeQoP2X0 Unless there's something blindingly obvious that is making a handful of people I've asked for help all blind to it.

**Comment by ZeroQI** [comment]
folders and content ok, could be right issue
sometimes windows have 2 plex folders and agents pasted in wrong location that seem genuine
nothing in Plex Media server/Logs/Plex Media scanner.log?

**Comment by Hezkezl** [comment]
> folders and content ok, could be right issue
> sometimes windows have 2 plex folders and agents pasted in wrong location that seem genuine
> nothing in Plex Media server/Logs/Plex Media scanner.log?

My `C:\Users\Jordan\AppData\Local\Plex Media Server\Logs` folder is completely empty. 

.... actually, as part of that, I think I might have figured it out. Going in to Plex and exporting the logs... I noticed that it was loading info for a media bundle for something on a completely different drive. Went snooping on that drive, and struck gold. Apparently at some point in the past, I moved my Plex data over to a completely different drive (probably to spare my SSD from all that reading/writing)

So I moved the HANA and ASS files into the appropriate folders there... and they now show up. Ffffffffffffuuuuuuuuuuu I had no idea I moved that...

Thank you all for the help, and I apologize for taking up your time in trying to figure this out! I knew it should've been something easy, and apparently I was a moron and had no idea that I'd separated my Plex data to be separate from my install folders... ><
